data-structures/queues-1.js

```diff
@@ -0,0 +1,24 @@
+export function getFront(queue) {
+  if (queue.length === 0) {
+    return undefined;
+  } else {
+    return queue.peek();
+  }
+}
+
+export function addToBack(queue, value) {
+  queue.enqueue(value);
+}
+
+export function takeFront(queue) {
+  if (queue.length === 0) {
+    return undefined;
+  } else {
+    return queue.dequeue();
+  }
+}
+
+export function isEmpty(queue) {
+  const topQueue = queue.peek();
+  return typeof topQueue === 'undefined';
```

Comment:
In modern JS there is no reason to use this `typeof` technique to check for undefined. Just use `return topQueue === undefined;`.